packages/eslint-plugin/src/rules/prefer-nullish-coalescing.ts

```diff
@@ -309,7 +309,7 @@ export default createRule<Options, MessageIds>({
       ): void {
         const tsNode = parserServices.esTreeNodeToTSNodeMap.get(node);
         const type = checker.getTypeAtLocation(tsNode.left);
-        if (!isNullableType(type)) {
+        if (!isTypeFlagSet(type, ts.TypeFlags.Null | ts.TypeFlags.Undefined)) {
```

Comment:
@Josh-Cena threading https://github.com/typescript-eslint/typescript-eslint/pull/8262#issuecomment-1895982325:

> I'm good with this solution 👍 I wonder though, shouldn't we recommend switching to nullish coalescing for any/unknown too, with an option to turn that off, just like primitives? Because any/unknown are also nullable, it doesn't make no sense to use nullish coalescing.

Hmm, good question. I think if the type is explicitly not known then it'd be risky to suggest making any concrete changes to its handling. I'd be in support of this as an opt-in in for v6 and _maybe_ an opt-out for v7.